### PR TITLE
follow-up pr

### PR DESCRIPTION
**Pull request by vabshere**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -43,25 +47,21 @@ import {AnalyticsDirective} from './directives/analytics.directive';
...
     IdeasComponent,
     LandingComponent,
+    RegisterComponent,
+    TncComponent,
     AnalyticsDirective,
   ],
   imports: [
     BrowserModule,
     BrowserAnimationsModule,
-    // material
```
Comment (MohitKS5) [inline src/app/app.module.ts]: why are these comments removed?

File: src/app/dashboard/components/leaderboard/leaderboard.component.ts [hunk 2]
```diff
@@ -1,15 +1,71 @@
...
+    // rank: rank.toString(),
+    id: id.toString(),
+    name: name,
+    progress: Math.round(Math.random() * 100).toString(),
+    color: COLORS[Math.round(Math.random() * (COLORS.length - 1))]
+  };
+}
+
+/** Constants used to fill up our data base. */
+const COLORS = ['maroon', 'red', 'orange', 'yellow', 'olive', 'green', 'purple',
```
Comment (MohitKS5) [inline src/app/dashboard/components/leaderboard/leaderboard.component.ts]: why is this needed ?? avoid making big PRs uselessly. remove colors and randomize functions , use simple string/number arrays intead

File: src/app/homepage/components/banner/banner.component.html [hunk 3]
```diff
@@ -15,7 +15,7 @@
           <a class="nav-link" href="#">Register</a>
         </li>
         <li class="nav-item">
-          <div class="nav-link" (click)="onhit()">Login</div>
+          <a class="nav-link" (click)="onhit()">Login</a>
```
Comment (MohitKS5) [inline src/app/homepage/components/banner/banner.component.html]: why did this became an anchor?

File: src/app/homepage/components/landing/landing.component.html [hunk 4]
```diff
@@ -3,4 +3,5 @@
 <app-about></app-about>
 <app-properties></app-properties>
 <app-resp></app-resp>
-<app-contact></app-contact>
\ No newline at end of file
+<app-contact></app-contact>
+<!-- <app-register></app-register> -->
```
Comment (MohitKS5) [inline src/app/homepage/components/landing/landing.component.html]: remove this line. Remove such useless comments before pushing

File: src/app/homepage/components/register/follow-portals/follow-portals.component.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (MohitKS5) [inline src/app/homepage/components/register/follow-portals/follow-portals.component.ts]: no empty components are required, remove this component.

File: src/app/homepage/components/register/register.component.html [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+          <td>
+            <mat-form-field class="full-width">
+              <input matInput [matDatepicker]="picker" placeholder="Birthdate" disabled>
+              <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
+              <mat-datepicker #picker disabled="false"></mat-datepicker>
+            </mat-form-field>
+          </td>        
+          <td>
+            <mat-form-field class="full-width">
+              <mat-select required ngModel #sex="ngModel" name="sex" placeholder="Select Sex">
```
Comment (MohitKS5) [inline src/app/homepage/components/register/register.component.html]: sex can never be a required thing ! either remove required or add an option not to specify

File: src/app/homepage/components/register/register.component.html [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+                <mat-error *ngIf="sex.errors.required">Required</mat-error>
+              </mat-select>
+            </mat-form-field>
+          </td>
+        </tr>
+        <tr>
+          <td>
+            <mat-form-field class="full-width">
+              <mat-select required ngModel #year="ngModel" name="year" placeholder="Select Year">
+                <mat-option *ngFor="let year of collegeYear" [value]="year.id">{{ collegeYear.name }}</mat-option>
```
Comment (MohitKS5) [inline src/app/homepage/components/register/register.component.html]: collegeYear.name is not defined, did you even check console??

File: src/app/homepage/components/register/register.component.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  styleUrls: ['./register.component.css']
+})
+export class RegisterComponent implements OnInit {
+  sexes = [
+    { id: 1, name: 'Male' },
+    { id: 2, name: 'Female' },
+    { id: 3, name: 'Other' },
+  ];
+
+  collegeYear = [
```
Comment (MohitKS5) [inline src/app/homepage/components/register/register.component.ts]: why does year names have genders in it??

File: src/app/homepage/components/register/tnc/tnc.component.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (MohitKS5) [inline src/app/homepage/components/register/tnc/tnc.component.ts]: again an empty component, can keep this one , fill it with lorem ipsum atleast and implement where you wanted to show

File: src/app/mat-components/mat-components.module.spec.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import { MatComponentsModule } from './mat-components.module';
```
Comment (MohitKS5) [inline src/app/mat-components/mat-components.module.spec.ts]: what does this component do ?

File: src/app/mat-components/mat-components.module.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import { NgModule } from '@angular/core';
```
Comment (MohitKS5) [inline src/app/mat-components/mat-components.module.ts]: if you are using separate material module then copy paste other material imported components from app.module also

**Comment by MohitKS5** [comment]
- you are making too big PRs, you have to solve conflicts if you do this . By the time you submit this long PR , I would have done other changes and you will have to solve conflicts.
- don't use rubbish functions and unprofessional things like "filter as you may"
- when you use mock data keep it as simple as possible to replace

File: src/app/app.module.ts (hunk 1 above)
Comment (vabshere) [reply src/app/app.module.ts]: first, these comments were ill-placed. second that i have created a separate module for using material components to make code cleaner and more understandable

File: src/app/homepage/components/banner/banner.component.html (hunk 3 above)
Comment (vabshere) [reply src/app/homepage/components/banner/banner.component.html]: the other nav-link's were anchors so i did it to make the code uniform. there was also an issue of the cursor not changing to pointer when hovered on login so this fixed it

File: src/app/mat-components/mat-components.module.ts (hunk 11 above)
Comment (vabshere) [reply src/app/mat-components/mat-components.module.ts]: i think i already have

File: src/app/homepage/components/register/register.component.html (hunk 6 above)
Comment (vabshere) [reply src/app/homepage/components/register/register.component.html]: i've already the option "other". still, i'll remove required.

File: src/app/homepage/components/banner/banner.component.html (hunk 3 above)
Comment (MohitKS5) [reply src/app/homepage/components/banner/banner.component.html]: Others have to be removed anyways , only login button would be there.  Also you didn't need to change it in a PR meant for leaderboard and profile

File: src/app/mat-components/mat-components.module.spec.ts (hunk 10 above)
Comment (vabshere) [reply src/app/mat-components/mat-components.module.spec.ts]: it's a testing file automatically created by cli
